From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct ipv6_pinfo
Patch-mainline: never, kabi
References: kabi

In networking-stable-17_12_31, commit
513674b5a2c9c7a67501506419da5c3c77ac6f08 (net: reevalulate autoflowlabel
setting after sysctl setting) added one bit to a bitmask in struct
ipv6_pinfo. It made the kABI checker to complain.

Given the bit did not change the structure size and autoflowlabels are
used in the ipv6 core code only, just hide the change from the kABI
checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/ipv6.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/linux/ipv6.h
+++ b/include/linux/ipv6.h
@@ -255,8 +255,12 @@ struct ipv6_pinfo {
 						 * 100: prefer care-of address
 						 */
 				dontfrag:1,
+#ifdef __GENKSYMS__
+				autoflowlabel:1;
+#else
 				autoflowlabel:1,
 				autoflowlabel_set:1;
+#endif
 	__u8			min_hopcount;
 	__u8			tclass;
 	__be32			rcv_flowinfo;
